src/main/java/net/minecraftforge/client/player/inventory/tabs/GuiTab.java

```diff
@@ -0,0 +1,605 @@
+            FontRenderer fontRenderer)
+    {
+
+        boolean isFirstRow = isTopRow();
+        int column = getColumn();
+        int texX = column * 28;
+        int texY = 0;
+        int x = guiLeft + 28 * column;
+        int y = guiTop;
+
+        if (isSelectedtab)
+        {
+            texY += 32;
+        }
+
+        if (isAlignedRight())
+        {
+            x = guiLeft + xSize - 28 * (6 - column);
+        }
+        else if (column > 0)
+        {
+            x += column;
+        }
+
+        if (isFirstRow)
+        {
+            y -= 28;
+        }
+        else
+        {
+            texY += 64;
+            y += (ySize - 5);
+        }
+        RenderHelper.enableGUIStandardItemLighting();
+        GlStateManager.disableLighting();
+        GlStateManager.color(1F, 1F, 1F); // Reset color in case Items change it.
+        GlStateManager.enableBlend(); // Make sure blend is enabled else tabs show a white border.
+        guiContainer.drawTexturedModalRect(x, y, texX, texY, 28, 32);
+        x += 6;
+        y += 8 + (isFirstRow ? 1 : -1);
+        GlStateManager.enableLighting();
+        if (!getIconStack().isEmpty())
+        {
+            ItemStack itemstack = getIconStack();
+            itemRender.renderItemAndEffectIntoGUI(itemstack, x, y);
+            itemRender.renderItemOverlays(fontRenderer, itemstack, x, y);
+        }
+        else if (getIconResLoc() != null)
+        {
+            Minecraft.getMinecraft().getTextureManager().bindTexture(getIconResLoc());
+            Gui.drawModalRectWithCustomSizedTexture(x, y, 0f, 0f, 16, 16, 16f, 16f);
+        }
+        GlStateManager.disableLighting();
+    }
+
+    public boolean isAlignedRight()
+    {
+        return this.getColumn() == 5;
+    }
+
+    /**
+     * Returns the index relative to the gui this tab was registered too
+     */
+    public int getTabIndex()
+    {
+        return tabRegistry.get(parentContainer).indexOf(this);
+    }
+
+    /**
+     * Whether this tab is on the top row or not
+     */
+    public boolean isTopRow()
+    {
+        return (getTabIndex() > 11 ? (getTabIndex() - 12) % 12 : this.getTabIndex()) < 6;
+    }
+
+    /**
+     * Returns index % 6 to determine column
+     */
+    public int getColumn()
+    {
+        return (getTabIndex() > 11 ? (getTabIndex() - 12) % 12 : this.getTabIndex()) % 6;
+    }
+
+    /**
+     * @return Whether this tab's parent GUI is the current GUI
+     */
+    public boolean isActiveTab(Class<? extends GuiContainer> guiContainer)
+    {
+        return guiContainer == getTargetGui();
+    }
+
+    /**
+     * @return The tabindex / 12
+     */
+    public int getTabPage()
+    {
+        return getTabIndex() / 12;
+    }
+
+    /**
+     * @return The localized name for this tab
+     */
+    public String getLocalizedName()
+    {
+        return "guiGroup." + name;
```

Comment:
Why "guiGroup" why not guiTab or something"

Reply:
so recommend to change this to `guiTab` ?
I used guiGroup as in reference of the creative tabs, but a change can be made

Reply:
I'd use `guiGroup.`.